### PR TITLE
Moved variant annotation hists functions to gnomad_hail

### PR DESCRIPTION
**Pull request by ch-kr**
from gnomad_qc (https://github.com/macarthur-lab/gnomad_qc/blob/master/gnomad_qc/v2/variant_qc/make_var_annot_hists.py)

### CODE REVIEW

File: gnomad_hail/utils/annotations.py [hunk 1]
```diff
@@ -419,3 +419,63 @@ def annotate_freq(
 
     # Return MT with freq row annotation
     return mt.annotate_rows(freq=freq_expr).drop('_freq_meta')
+
+
+def get_annotations_hists(
+    ht: hl.Table, annotations_hists: Dict[str, Tuple],
```
Comment (lfrancioli) [inline gnomad_hail/utils/annotations.py]: I think moving `annotations_hists` to the next line would increase readability

File: gnomad_hail/utils/annotations.py [hunk 2]
```diff
@@ -419,3 +419,63 @@ def annotate_freq(
...
+    log10_annotations: List[str] = ['DP']
+    ) -> Dict[str, hl.expr.StructExpression]:
+    """
+    Creates histograms for variant metrics.
```
Comment (lfrancioli) [inline gnomad_hail/utils/annotations.py]: How about "Creates histograms for variant metrics in `ht.info` ?"

File: gnomad_hail/utils/annotations.py [hunk 3]
```diff
@@ -419,3 +419,63 @@ def annotate_freq(
...
+    :param Table ht: Table with variant metrics
+    :param Dict annotations_hists: Dictionary of metrics names and their histogram values (start, end, bins)
+    :param List log10_annotations: List of metrics to log scale
+    :return: Dictionary of metrics and their histograms
+    :rtype: Dict[str, hl.expr.StructExpression]
+    """
+    hist_dict = {}
+
+    # Check all fields in ht.info and create histograms if they are in annotations_hists dict
+    info_fields = list(ht.row.info)
```
Comment (lfrancioli) [inline gnomad_hail/utils/annotations.py]: unnecessary: you can directly write `for field in ht.row.info:`

Comment (lfrancioli) [reply gnomad_hail/utils/annotations.py]: Also -and up to you if you like/dislike it- but you could write this like that if you want:
```
return {
    field: hl.agg.hist(
            hl.log10(ht[field] if field in log10_annotations else ht[field), start, end, bin
    ) for field, (start, end, bins) in annotations_hists.items() if field in ht.row.info
}
```

File: gnomad_hail/utils/annotations.py [hunk 4]
```diff
@@ -419,3 +419,63 @@ def annotate_freq(
...
+            start, end, bins = annotations_hists[field]
+            if field in log10_annotations:
+                hist_dict[field] = hl.agg.hist(hl.log10(ht[field]), start, end, bins)
+            else:
+                hist_dict[field] = hl.agg.hist(ht[field], start, end, bins)
+
+    return hist_dict
+
+
+def create_frequency_bins(ht: hl.Table) -> hl.Table:
```
Comment (lfrancioli) [inline gnomad_hail/utils/annotations.py]: How about changing signature to be the following?
```
def create_frequency_bins_expr(
   ac: hl.expr.NumericExpression,
   af:  hl.expr.NumericExpression
) -> hl.expr.StringExpression:
```

File: gnomad_hail/utils/annotations.py [hunk 5]
```diff
@@ -419,3 +419,63 @@ def annotate_freq(
...
+    """
+    Creates bins for frequencies in preparation for aggregating QUAL by frequency bin.
```
Comment (lfrancioli) [inline gnomad_hail/utils/annotations.py]: I think it's worth adding the list of bins created

File: gnomad_hail/utils/annotations.py (hunk 1 above)
Comment (nawatts) [reply gnomad_hail/utils/annotations.py]: Plug for [Black](https://pypi.org/project/black/) to automatically resolve all formatting discussions.

File: gnomad_hail/utils/annotations.py (hunk 3 above)
Comment (ch-kr) [reply gnomad_hail/utils/annotations.py]: ooh, nice!

**Comment by ch-kr** [comment]
Thanks for the suggestions! Back to you

**Pull request merged**